jetty-http/src/main/java/org/eclipse/jetty/http/HttpURI.java

```diff
@@ -1281,6 +1282,21 @@ private void checkSegment(String uri, int segment, int end, boolean param)
         {
             if (!_ambiguous.contains(Ambiguous.SEGMENT))
             {
+                if (segment == end)
+                {
+                    // Falsely reported empty segment before the first slash.
+                    if (segment == 0)
+                        return;
+
+                    // If the segment starts after the URI string it means the string ended with a '/'.
+                    if (segment > uri.length() - 1)
+                        return;
+
+                    // It is only a true empty segment if it actually ends with a '/' character.
+                    if (uri.charAt(end) != '/')
+                        return;
+                }
```

Comment:
Is is possible to avoid calling checkSegment if there really isn't a segment?

Reply:
Hmmm if I comment this section out, all the tests still pass in HttpURITest, so at the very least it needs coverage